### ISSUE TITLE
Confusing message when string ends with a backslash

### ISSUE THREAD

**Issue by bkoelman** [fluentassertions/fluentassertions#1018]
Consider the next example:
```csharp
string actual = @"some\";
string expected = @"some";
actual.Should().Be(expected);
```
The test fails with message:
```
Expected actual to be "some" with a length of 4, but "some" has a length of 5, differs near "" (index 4).
```

Observed in FluentAssertions v5.6.0, running on .NET Core 2.0.

**Comment by dennisdoomen** [comment]
@jnyrup isn't this fixed by #987?

**Comment by jnyrup** [comment]
@dennisdoomen Indeed it is.

From the next release the error message is
```
Expected actual to be "some" with a length of 4, but "some\" has a length of 5, differs near "\" (index 4).
```
